examples/main.cpp

```diff
@@ -131,9 +134,9 @@ void sample2_mlp(const string& data_dir) {
 #if defined(_MSC_VER) && _MSC_VER < 1800
   // initializer-list is not supported
   int num_units[] = {28 * 28, num_hidden_units, 10};
-  auto nn         = make_mlp<tan_h>(num_units, num_units + 3);
+  auto nn         = make_mlp<tanh_layer>(num_units, num_units + 3);
```

Comment:
Because `make_mlp` needs to keep activations after each fully connected layer. Hence we specify an activation layer as template parameter now.
so this makes
```
nn << fc(num_units) << tanh() << fc(num_units + 3) << tanh()
```